app/root.tsx

```diff
@@ -307,6 +307,26 @@ function PageLoadingMessage() {
   )
 }
 
+declare global {
+  const fathom: {
+    trackPageview(): void
+  }
+}
+
+function CanonicalLink({origin, path}: {origin: string; path: string}) {
+  const matches = useMatches()
+  const leafRoute = matches[matches.length - 1]
+  const canonicalUrl = removeTrailingSlash(
```

Comment:
Should this not just be `useLocation().pathname`?

Reply:
Actually yes, in my case I was using `useMatches` because I wanted to leave the option to change the canonical URL via a loader or handle, but if you don't need this then `useLocation` is fine. Fixed.